controllers/sharhSearch.controller.js

```diff
@@ -140,7 +123,10 @@ class SharhSearchController {
 
     const sharhId = tabElement.querySelector('a[xplain]')?.getAttribute('xplain');
     if (!sharhId) {
-      throw new AppError('No sharh found for the given text', 404);
+      return sendSuccess(res, 200, [], {
+        specialist: req.isForSpecialist,
+        isCached: false,
+      });
```

Comment:
This is not a collection, this is a single resource so we can throw an error if it is not exist